pkg/util/pointer.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+// GetValueOrDefault returns the value pointed to by p, or a zero value of
+// its type if p is nil.
+func GetValueOrDefault[T any](p *T) T {
```

Comment:
Would it be more appropriate to name it `GetOrZero`.

Reply:
Agree